src/mlpack/bindings/python/tests/test_python_binding.py

```diff
@@ -1336,5 +1336,161 @@ def testModelForceCopy(self):
     self.assertEqual(output2['model_bw_out'], 20.0)
     self.assertEqual(output3['model_bw_out'], 20.0)
 
+  def testCheckInputMatricesNaN(self):
+    """
+    Checks that an exception is thrown if the input matrix contains
+    NaN values.
+    """
+    x = np.random.randint(low=0, high=500, size=[100, 5]).astype(float)
+    a = np.random.randint(low=0, high=100)
+    b = np.random.randint(low=0, high=5)
+    x[a][b] = np.nan
+    self.assertRaises(RuntimeError,
+                      lambda : test_python_binding(string_in="hello",
+                                                   int_in=12,
+                                                   double_in=4.0,
+                                                   mat_req_in=[[1.0]],
+                                                   col_req_in=[1.0],
+                                                   matrix_in=x,
+                                                   check_input_matrices=True))
+
+    self.assertRaises(RuntimeError,
+                      lambda : test_python_binding(string_in="hello",
+                                                   int_in=12,
+                                                   double_in=4.0,
+                                                   mat_req_in=[[1.0]],
+                                                   col_req_in=[1.0],
+                                                   umatrix_in=x,
```

Comment:
Actually, this one might not work because there is no way to represent NaN or Inf with an `arma::Mat<size_t>`.  So I think perhaps the checks are only necessary for double-type matrices or vectors.